cmd/server/controllers/SellerController.go

```diff
@@ -0,0 +1,143 @@
+	Address     string `json:"address" binding:"required"`
+	Telephone   string `json:"telephone" binding:"required"`
+}
+
+type requestSellerPatch struct {
+	Address   string `json:"address" binding:"required"`
+	Telephone string `json:"telephone" binding:"required"`
+}
+
+type SellerController struct {
+	service seller.Service
+}
+
+func NewSeller(s seller.Service) SellerController {
+	return SellerController{
+		service: s,
+	}
+}
+
+func (c SellerController) GetAll() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		seller, err := c.service.GetAll()
+		if err != nil {
+			ctx.JSON(http.StatusBadRequest, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+
+		ctx.JSON(http.StatusOK, gin.H{
+			"data": seller,
+		})
+	}
+}
+
+func (c SellerController) GetById() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		id, err := strconv.ParseInt(ctx.Param("id"), 10, 64)
+		if err != nil {
+			ctx.JSON(http.StatusInternalServerError, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+		seller, err := c.service.GetById(id)
+		if err != nil {
+			ctx.JSON(http.StatusNotFound, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+		ctx.JSON(http.StatusOK, gin.H{
+			"data": seller,
+		})
+	}
+}
+
+func (c SellerController) CreateSeller() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		var req requestSellerPost
+		if err := ctx.ShouldBindJSON(&req); err != nil {
+			ctx.AbortWithStatusJSON(http.StatusUnprocessableEntity, gin.H{
+				"error":   err.Error(),
+				"message": "invalid request",
+			})
+			return
+		}
+		seller, err := c.service.CreateSeller(req.Cid, req.CompanyName, req.Address, req.Telephone)
+		if err != nil {
+			ctx.JSON(http.StatusConflict, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+		ctx.JSON(http.StatusCreated, gin.H{
+			"data": seller,
+		})
+	}
+}
+
+func (c SellerController) UpdateSellerAddresAndTel() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+
+		id, err := strconv.ParseInt(ctx.Param("id"), 10, 64)
+		if err != nil {
+			ctx.JSON(http.StatusInternalServerError, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+		var req requestSellerPatch
+		if err := ctx.ShouldBindJSON(&req); err != nil {
+			ctx.AbortWithStatusJSON(http.StatusUnprocessableEntity, gin.H{
+				"error":   err.Error(),
+				"message": "invalid request",
+			})
+			return
+		}
+		seller, err := c.service.UpdateSellerAddresAndTel(id, req.Address, req.Telephone)
+		if err != nil {
+			ctx.JSON(http.StatusNotFound, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+		ctx.JSON(http.StatusOK, gin.H{
+			"data": seller,
+		})
+	}
+}
+
+func (c SellerController) DeleteSeller() gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		id, err := strconv.ParseInt(ctx.Param("id"), 10, 64)
+		if err != nil {
+			ctx.JSON(http.StatusInternalServerError, gin.H{
+				"error": err.Error(),
+			})
+			return
+		}
+		err = c.service.DeleteSeller(id)
+		if err != nil {
+			ctx.JSON(http.StatusNotFound, gin.H{})
```

Comment:
Aqui quando ele não encontrar, tem que manter a msg que estava, de not found

Reply:
👍🏽